### DISCUSSION TITLE
serve local lan addresses?

### DISCUSSION THREAD

**Discussion by airdogvan** [DNSCrypt/dnscrypt-proxy#2387] [Q&A]
I'm trying to figure out how to make it serve local addresses (I.e. next.local.joe points to 192.168.1.03), googled my head off and couldn't find an answer.

Does this mean that dnscrypt does not do that?

I will gladly of course accept a "no", it's just that I'm surprised as this seems to me pretty basic functionality.

**Comment by ghost** [comment]
[Cloaking](https://github.com/DNSCrypt/dnscrypt-proxy/wiki/Cloaking) or [Forwarding](https://github.com/DNSCrypt/dnscrypt-proxy/wiki/Forwarding), whatever you prefer.

**Comment by airdogvan** [comment]
Thanks for the quick answer.

In my case it would be cloaking as forwarding would work if I had another local DNS server for local addresses.

So I tried cloaking and it works on the local (dnscrypt) machine.  When I do a dnscrypt-proxy -resolve home.local.joe it returns the address with the mention: This query has been locally blocked by dnscrypt-proxy.  

Which I took to mean that it doesn't forward the request to outside servers.

But then when I do a dig home.local.joe @IP_dnscrypt_machine I get:


dig home.local.joe  @192.168.1.03 (IP for dnscrypt)

; <<>> DiG 9.18.12-0ubuntu0.22.04.1-Ubuntu <<>> home.local.joe @192.168.1.03
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NXDOMAIN, id: 6997
;; flags: qr rd ra; QUERY: 1, ANSWER: 0, AUTHORITY: 1, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 512
;; QUESTION SECTION:
;home.local.joe.              IN      A

;; AUTHORITY SECTION:
.                       457     IN      SOA     a.root-servers.net. nstld.verisign-grs.com. 2023050800 1800 900 604800 86400

;; Query time: 0 msec
;; SERVER: 192.168.1.03#53(192.168.1.03) (UDP)
;; WHEN: Mon May 08 09:22:00 PDT 2023
;; MSG SIZE  rcvd: 120


And of course I can't ping that home.local.joe let alone connect to it...


I've searched the config file for blocking options but apart from not forwarding the request to other servers (## Immediately respond to queries for local zones instead of leaking them to
## upstream resolvers (always causing errors or timeouts).

block_undelegated = true
)

I didn't find anything else that might prevent it from answering those local requests....

**Comment by ghost** [comment]
DNSCrypt-Proxy is a client, not a server.

However `This query has been locally blocked by dnscrypt-proxy` means that the IP was intentionally _not_ resolved, because a user setting prevented it.

**Comment by jedisct1** [comment]
If you get "This query has been locally blocked by dnscrypt-proxy", there's indeed a rule in your configuration that blocks it.

This is not specific to dnscrypt-proxy, but if you're new to something, start with the default configuration, and change things incrementally, one by one, so you understand what they do.

You can also enable the query log, and it will tell you why the query is blocked.

**Comment by airdogvan** [comment]
Ok it's working now.  Problem is I'm not sure why.

And thanks for the help guys, much appreciated.

And no not really new at this, have been doing this for the last 30 years....

New to dnscrypt though.

Anyway thanks again, much appreciated.
